src/sentry/static/sentry/app/utils/apm.jsx

```diff
@@ -2,39 +2,46 @@ import * as Router from 'react-router';
 import * as Sentry from '@sentry/browser';
 
 let firstPageLoad = true;
+let flushTransactionTimeout = undefined;
+let wasInterrupted = false;
+
+const TRANSACTION_TIMEOUT = 5000;
+const requests = new Set([]);
+const renders = new Set([]);
+const hasActiveRequests = () => requests.size > 0;
+const hasActiveRenders = () => renders.size > 0;
 
 function startTransaction() {
   // We do set the transaction name in the router but we want to start it here
   // since in the App component where we set the transaction name, it's called multiple
   // times. This would result in losing the start of the transaction.
   Sentry.configureScope(scope => {
     if (firstPageLoad) {
-      firstPageLoad = false;
-    } else {
-      const prevTransactionSpan = scope.getSpan();
-      // If there is a transaction we set the name to the route
-      if (prevTransactionSpan && prevTransactionSpan.timestamp === undefined) {
-        prevTransactionSpan.finish();
-      }
-      scope.setSpan(
-        Sentry.startSpan({
-          op: 'navigation',
-          sampled: true,
-        })
-      );
+      return;
     }
-  });
 
-  finishTransaction(5000);
-}
+    // If there's a previous span open, finish it
+    // TODO(apm): I think this is wrong, we probably only want to finish the spans
+    // that we start in this function and in `startApm()`
+    //
+    // i.e. it could close out a span of an API request
+    const prevTransactionSpan = scope.getSpan();
+    if (prevTransactionSpan) {
+      prevTransactionSpan.finish();
+    }
 
-const requests = new Set([]);
-const renders = new Set([]);
-let flushTransactionTimeout = undefined;
-let wasInterrupted = false;
+    scope.setSpan(
+      Sentry.startSpan({
+        op: 'navigation',
```

Comment:
Should we define a colour for this in the known op types?

Reply:
I'm not sure, this is going to be the root span I believe, which seems to always be the same color?